tests/testsuite/update_duplicated_with_precise_breaking_feature.rs

```diff
@@ -0,0 +1,712 @@
+//! Duplicating tests for `cargo update --precise` with the
+//! update-precise-breaking feature enabled. When the feature is stabilized,
+//! this file can be deleted.
+
+#![allow(deprecated)]
+
```

Comment:
Still trying to decide what the best approach is here but in the mean time, could you split this into two commits
- One that duplicates the tests
- One that enables the unstable feature

That way it shows what behavior change happened (or not) with the feature.

Reply:
Instead of duplicating tests, one approach we could take is using a test only env var to activate the unstable behavior, for example [`__CARGO_USE_GITOXIDE_INSTEAD_OF_GIT2`](https://doc.crates.io/contrib/tests/running.html#running-with-gitoxide-as-default-git-backend-in-tests).

(Yes I admit that this opens a door for people without `-Zunstable-options`, though we already have `__CARGO_TEST_CHANNEL_OVERRIDE_DO_NOT_USE_THIS` hence 😬)

Reply:
`__CARGO_USE_GITOXIDE_INSTEAD_OF_GIT2` is used from tests to modify Cargo behaviour. You would still need duplicate tests if you wanted to run Cargo with both options.

I was thinking the duplicate test file is OK, as it is meant to be temporary and can be deleted later.

I have put it in a separate commit, and it didn't need any modification after implementing the feature.

Reply:
This was split out into its own commit but I'd like to see it split into two commits.

Currently, the commit is
```console
$ cp tests/testsuite/update.rs
$ edit tests/testsuite/update_duplicated_with_precise_breaking_feature.rs
$ git commit
```
I was asking to see
```console
$ cp tests/testsuite/update.rs
$ git commit
$ edit tests/testsuite/update_duplicated_with_precise_breaking_feature.rs
$ git commit
```
Depending on how well it diffs, maybe it should be more

This makes it easier to audit in this review, and in the future, what the differences are between the two tests files.